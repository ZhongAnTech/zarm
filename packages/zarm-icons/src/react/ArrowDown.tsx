import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowDown = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M494.094 733.426a41.472 41.472 0 01-20.023-11.126L140.16 388.388c-16.272-16.271-16.272-42.653 0-58.925 16.271-16.272 42.653-16.272 58.925 0L503.66 634.037 808.186 329.51c16.271-16.272 42.653-16.272 58.925 0s16.272 42.654 0 58.926L533.2 722.347c-10.607 10.607-25.508 14.3-39.106 11.08z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowDown);
export default ForwardRef;
