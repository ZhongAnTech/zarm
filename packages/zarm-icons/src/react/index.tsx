export { default as ArrowDown } from './ArrowDown';
export { default as ArrowLeft } from './ArrowLeft';
export { default as ArrowRight } from './ArrowRight';
export { default as ArrowUp } from './ArrowUp';
export { default as CloseCircleFill } from './CloseCircleFill';
export { default as CloseCircle } from './CloseCircle';
export { default as Close } from './Close';
export { default as DeleteKey } from './DeleteKey';
export { default as Empty } from './Empty';
export { default as Keyboard } from './Keyboard';
export { default as Minus } from './Minus';
export { default as PlusCircle } from './PlusCircle';
export { default as Plus } from './Plus';
export { default as Search } from './Search';
export { default as SuccessCircle } from './SuccessCircle';
export { default as Success } from './Success';
export { default as Volume } from './Volume';
export { default as WarningCircle } from './WarningCircle';
export { default as Warning } from './Warning';
