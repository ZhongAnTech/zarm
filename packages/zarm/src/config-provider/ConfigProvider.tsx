import React, { PureComponent, createContext } from 'react';
import type { Context } from 'react';
import type { ConfigProviderProps, Locale } from './PropsType';
import setTheme from './setTheme';
import setPrimaryColor from './setPrimaryColor';

const defaultConfig: ConfigProviderProps = {
  locale: ({} as unknown) as Locale,
  theme: 'light',
  primaryColor: '#00bc70',
};

export const LocaleContext: Context<Locale> = createContext(defaultConfig.locale);
export const ConfigContext: Context<ConfigProviderProps> = createContext(defaultConfig);

let runTimeLocale: Locale;
const changeRunTimeLocale = (locale: Locale) => {
  runTimeLocale = locale;
};

export const getRunTimeLocale = () => runTimeLocale;

export default class ConfigProvider extends PureComponent<ConfigProviderProps, {}> {
  static defaultProps = defaultConfig;

  componentDidMount() {
    this.update();
  }

  componentDidUpdate() {
    this.update();
  }

  update() {
    const { locale, theme, primaryColor } = this.props;
    changeRunTimeLocale(locale);
    setTheme(theme);
    setPrimaryColor(primaryColor);
  }

  render() {
    const { locale, theme, primaryColor, children } = this.props;
    return (
      <ConfigContext.Provider value={{ locale, theme, primaryColor }}>
        {React.Children.only(children)}
      </ConfigContext.Provider>
    );
  }
}
