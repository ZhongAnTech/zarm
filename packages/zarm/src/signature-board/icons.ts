export const DEL_ICON =
  'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAMAAABHPGVmAAAAflBMVEUAAABRUVFRUVFRUVFPT09PT09QUFBQUFBKSkpRUVFPT09QUFBRUVFSUlJRUVFRUVFRUVFSUlJRUVFQUFBRUVFRUVFRUVFRUVFQUFBRUVFRUVFRUVFRUVFRUVFRUVFRUVFRUVFRUVFRUVFSUlJPT09QUFBRUVFVVVVYWFhTU1ORL5V1AAAAJnRSTlMAA/r8BSqAuQ7xR0a3fULqqBPzCuXMXVIZo5qFbzbfwXho1zAikA+znhcAAAKfSURBVGje7ZrZjqMwEEXLS8ja7EtCtl5d9v//4IB7JAgjdWFgeqYln0enrg8FETg44PlxMMlmltAKh8LpDhk+nQ7rrzkF+2yypclVhdFIos2tnN7MBxolxmCwhomctBBqFELoekorDC6a2wk4gbBV+DLFkuTGxlEToBKNxOSJs0LCBVuHSeOA4MRNa8FLE3LlbiXpG5BcI3u+7uBMlps2eQZJAkF7PCbPnCWbrWowO/ocMKiMathuJkrUHhgt2at5kt0PlLAHOolkBBJ2vyWDOeh7tfyUhDCC8FMiyUmz1SPX1CbL9xXJe2lL0+tgPBs6niM+QFn4F1C1UfDYVqmNWhyjK2A9SYx8eQnH+Jsk33e6ugu/OFFAfIWXIHNYOC25bGOLA/8eJv9Y3kopYcwQcxMRQ8TnNPuP82oQKetL0h9ikFzqcuA4vlzexrZwRtSD526sUb8mDxO+aq3jh9w1QjQj13jHVHGORT++Q8G5DkD2jkRzLkzYn/LeDJkoGdVIqFqiTW+hF2iuBB6gxxqF4vqp80Jml1/KeknJzkrSUZKgJ9mkNrhzkGwdO+kWUV7iJV7iJV7iJV7iJV7iJV7iJX9RcraSNfEjaKbkBbniWAPrqk52qJwt6UhyrdBc+5IQUelbVzRbwuDtEBUVPFAV0WEFbDFJG9vAADvEFumks3TZ/tB8CY2XeMm3SnadhHwVlSaukmNqg+EoyTtXDSYE6QSExkquVkKR3YxQwhSJYyOF3cbdHoHG3sXtZu6tDhyoc5vqNhqoi2IPSRjULthQ91aStMSatwHBHRCqgesT6ehex2ITckZwLDbjX24ndzTCGYPF0eUPIvI5Nxqd0CZ/ln0HbYGkOsdrB+Jz1T0px2to5oeYZG7l/8OmjGcsvwB+KxB3VpOtpgAAAABJRU5ErkJggg==';

export const UNDO_ICON =
  'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAMAAABHPGVmAAAAhFBMVEUAAAAsLCwnJycsLCwsLCwlJSUsLCwyMjIsLCwsLCwsLCwsLCwsLCwrKyssLCwsLCwsLCwsLCwsLCwsLCwsLCwpKSksLCwrKyssLCwrKyssLCwsLCwsLCwrKyssLCwsLCwsLCwsLCwrKystLS0sLCwrKyssLCwqKiosLCwsLCwsLCwuLi7xqpc4AAAAKnRSTlMA/AOcYw7yBfo5dvZtSdOjXePakk8J6ooyEe+0rZgifkM+KhzJtykYysZAy40uAAAB+UlEQVRo3u3Y247aQAyAYTscEhJCQgiQcNou7KG13//96tHOwEVVElf1Rav579En4yA5QCwWi8Vi/1kJ/NI/anyk1lAKXXG2HUeMRUG8sVOCgWiqiJEREjnFZ2ZYKg/DKWeQbA3iHCRjYwmSuZFE4/fG690gZ6TJs/6CkdUwmEBq49vDQGo2kyft87r81DNJMEI8EB3e804+pzPQGz4ciIgZq4uCSaB0hi6BeL7vITUyQsI0V0jGGa039GVc1F4ZmqOgAnWRpPrV1qyPEB9KOaxcNtV5ouv0Y0fiBGV3GVRS+IP6tkIOCq+2w0tRB67uyPft5zYnh5P2XkFa92CTKMug8EsYxYDZM/pRZmCTIMkqKK3hKOXjeAqIgfLO+PV9bcEoQXKHuDrD72tBvtIKkT4Lv5TaENmuPZJbIg2hR1Kwqp/7SaaGi79m9FVriNThEV4YIhOPzG9gkxi3nV/JEYwSpGI03nsKLfmKiw0ixseB/CAnGyMBeN3dL5ZOg2hOiZy8gVzZDNKXbxwMyq4aJK2nY8r3px0xkk/5aKVrHhlRMDLtqZKuKFNe3BmfQNIhqH1DqQASSwTFeHGGDRKIpnWGCYIuYm6WWzUhSMM0Jpbmx+nMjaFH3rLDfLDD+vuxmt7AE+pmo9om5v9Qh/eTWCwWi8VisdizfgJaxKDFfwfcTwAAAABJRU5ErkJggg==';
