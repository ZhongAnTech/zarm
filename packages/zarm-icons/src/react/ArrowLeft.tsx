import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowLeft = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M296.114 508.035c-3.22-13.597.473-28.499 11.079-39.105l333.912-333.912c16.271-16.272 42.653-16.272 58.925 0s16.272 42.654 0 58.926L395.504 498.47l304.574 304.574c16.272 16.272 16.272 42.654 0 58.926s-42.654 16.272-58.926 0L307.241 528.058a41.472 41.472 0 01-11.127-20.023z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowLeft);
export default ForwardRef;
