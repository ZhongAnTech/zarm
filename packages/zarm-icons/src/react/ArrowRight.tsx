import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowRight = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M711.156 508.035a41.472 41.472 0 01-11.126 20.023L366.12 861.97c-16.272 16.272-42.654 16.272-58.926 0s-16.272-42.654 0-58.926L611.767 498.47 307.241 193.944c-16.272-16.272-16.272-42.654 0-58.926 16.271-16.272 42.653-16.272 58.925 0L700.078 468.93c10.606 10.606 14.299 25.508 11.078 39.105z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowRight);
export default ForwardRef;
