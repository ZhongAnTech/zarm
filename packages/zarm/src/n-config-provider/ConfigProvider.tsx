import * as React from 'react';
import type { Context } from 'react';
import type { ConfigProviderProps } from './interface';
import defaultLocaleData from './locale/zh_CN';
import setCssVars, { defaultCssVars } from './setCssVars';
import setPrimaryColor from './setPrimaryColor';
import setTheme from './setTheme';

const defaultConfig: ConfigProviderProps = {
  prefixCls: 'za',
  locale: defaultLocaleData,
  theme: 'light',
  safeIphoneX: false,
  cssVars: defaultCssVars,
};

export const ConfigContext: Context<ConfigProviderProps> = React.createContext(defaultConfig);

let runtimeConfigContext: ConfigProviderProps;

const changeRuntimeConfigContext = (props: ConfigProviderProps) => {
  runtimeConfigContext = props;
};

const ConfigProvider: React.FC<ConfigProviderProps> = (props) => {
  const { children, cssVars, primaryColor, theme, ...rest } = props;

  changeRuntimeConfigContext(props);

  React.useEffect(() => {
    primaryColor && setPrimaryColor(primaryColor);
  }, [primaryColor]);

  React.useEffect(() => {
    theme && setTheme(theme);
  }, [theme]);

  const newChildren = setCssVars(children, cssVars);

  return (
    <ConfigContext.Provider value={rest}>{React.Children.only(newChildren)}</ConfigContext.Provider>
  );
};

export const RuntimeConfigProvider: React.FC = ({ children }) => {
  const props = React.useRef(runtimeConfigContext);

  React.useEffect(() => {
    props.current = runtimeConfigContext;
  }, [runtimeConfigContext]);

  return <ConfigProvider {...props?.current}>{children}</ConfigProvider>;
};

ConfigProvider.displayName = 'ConfigProvider';
ConfigProvider.defaultProps = defaultConfig;

export default ConfigProvider;
