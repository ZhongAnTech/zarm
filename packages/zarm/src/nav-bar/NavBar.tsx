import React, { forwardRef } from 'react';
import type { HTMLAttributes } from 'react';
import classnames from 'classnames';
import { ConfigContext } from '../n-config-provider';
import type { BaseNavBarProps } from './interface';

export type NavBarProps = BaseNavBarProps & Omit<HTMLAttributes<HTMLDivElement>, 'title'>;

const NavBar = forwardRef<HTMLDivElement, NavBarProps>((props, ref) => {
  const { className, title, left, right, ...restProps } = props;

  const { prefixCls: globalPrefixCls } = React.useContext(ConfigContext);
  const prefixCls = `${globalPrefixCls}-nav-bar`;

  const cls = classnames(prefixCls, className);
  const titleCls = `${prefixCls}__title`;
  const sideCls = `${prefixCls}__side`;
  const leftCls = `${sideCls} ${prefixCls}__side--left`;
  const rightCls = `${sideCls} ${prefixCls}__side--right`;

  return (
    <div ref={ref} className={cls} {...restProps}>
      <div className={leftCls}>{left}</div>
      <div className={titleCls}>{title}</div>
      <div className={rightCls}>{right}</div>
    </div>
  );
});

NavBar.displayName = 'NavBar';

NavBar.defaultProps = {};

export default NavBar;
