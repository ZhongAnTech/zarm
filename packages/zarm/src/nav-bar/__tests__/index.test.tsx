import React from 'react';
import { render, screen } from '@testing-library/react';
import { ArrowLeft, ArrowRight } from '@zarm-design/icons';
import NavBar from '../index';
import ConfigProvider from '../../config-provider';

describe('NavBar', () => {
  it('should renders correctly', () => {
    const { asFragment } = render(<NavBar />);
    expect(asFragment().firstChild).toMatchSnapshot();
  });

  it('should renders correctly with title', () => {
    const { asFragment } = render(<NavBar title="这是标题" />);
    expect(asFragment().firstChild).toMatchSnapshot();
  });

  it('renders correctly with left content', () => {
    const { asFragment } = render(<NavBar left={<ArrowLeft />} />);
    expect(asFragment().firstChild).toMatchSnapshot();
  });

  it('renders correctly with right content', () => {
    const { asFragment } = render(<NavBar right={<ArrowRight />} />);
    expect(asFragment().firstChild).toMatchSnapshot();
  });

  it('should render with custom prefix className ', () => {
    render(
      <ConfigProvider prefixCls="custom">
        <NavBar data-testid="navbar" />
      </ConfigProvider>,
    );
    expect(screen.queryByTestId('navbar')).toHaveClass('custom-nav-bar');
  });

  test('should add HTML attributes to root element', () => {
    const { asFragment } = render(<NavBar id="a" />);
    expect(asFragment().firstChild).toHaveAttribute('id', 'a');
  });
});
