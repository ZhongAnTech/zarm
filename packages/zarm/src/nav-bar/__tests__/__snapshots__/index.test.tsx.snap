// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly with left content 1`] = `
<div
  class="za-nav-bar"
>
  <div
    class="za-nav-bar__side za-nav-bar__side--left"
  >
    <i
      class="za-icon"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 1000 1000"
        width="1em"
      >
        <path
          d="M296.114 508.035c-3.22-13.597.473-28.499 11.079-39.105l333.912-333.912c16.271-16.272 42.653-16.272 58.925 0s16.272 42.654 0 58.926L395.504 498.47l304.574 304.574c16.272 16.272 16.272 42.654 0 58.926s-42.654 16.272-58.926 0L307.241 528.058a41.472 41.472 0 01-11.127-20.023z"
          fill-rule="evenodd"
        />
      </svg>
    </i>
  </div>
</div>
`;

exports[`NavBar renders correctly with right content 1`] = `
<div
  class="za-nav-bar"
>
  <div
    class="za-nav-bar__side za-nav-bar__side--right"
  >
    <i
      class="za-icon"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 1000 1000"
        width="1em"
      >
        <path
          d="M711.156 508.035a41.472 41.472 0 01-11.126 20.023L366.12 861.97c-16.272 16.272-42.654 16.272-58.926 0s-16.272-42.654 0-58.926L611.767 498.47 307.241 193.944c-16.272-16.272-16.272-42.654 0-58.926 16.271-16.272 42.653-16.272 58.925 0L700.078 468.93c10.606 10.606 14.299 25.508 11.078 39.105z"
          fill-rule="evenodd"
        />
      </svg>
    </i>
  </div>
</div>
`;

exports[`NavBar should renders correctly 1`] = `
<div
  class="za-nav-bar"
/>
`;

exports[`NavBar should renders correctly with title 1`] = `
<div
  class="za-nav-bar"
>
  <div
    class="za-nav-bar__title"
  >
    这是标题
  </div>
</div>
`;
